connectors/byoc.py

```diff
@@ -601,8 +579,8 @@ async def validate_filtering(self, validator):
                 if validation_result.state == FilteringValidationState.VALID:
                     filter_["active"] = filter_.get("draft")
 
-        self.filtering = filtering
         await self.index.update(doc_id=self.id, doc={"filtering": filtering})
+        await self.reload()
```

Comment:
Same - do we actually want to reload everywhere then?

Reply:
Not everywhere. We try not to do so if possible. But this place is critical. Imagine a customer edits filtering in Kibana, BE validates it, and updates the active filtering. If we don't reload here, the connector will still hold the out-dated filtering, which will be used to schedule a new sync job.